code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -562,32 +549,26 @@ func (ssc *StorageSmartContract) GetAllocationMinLockHandler(ctx context.Context
 	creationDate := time.Now()
 
 	allocData := params.Get("allocation_data")
-	var request newAllocationRequest
-	if err = request.decode([]byte(allocData)); err != nil {
+	var req newAllocationRequest
+	if err = req.decode([]byte(allocData)); err != nil {
 		return "", common.NewErrInternal("can't decode allocation request", err.Error())
 	}
 
-	var allBlobbersList *StorageNodes
-	allBlobbersList, err = ssc.getBlobbersList(balances)
+	blobbers, err := ssc.getBlobbersForRequest(req, balances)
 	if err != nil {
-		return "", common.NewErrInternal("can't get blobbers list", err.Error())
+		return "", common.NewErrInternal("error selecting blobbers", err.Error())
 	}
-	if len(allBlobbersList.Nodes) == 0 {
-		return "", common.NewErrInternal("can't get blobbers list",
-			"no blobbers found")
-	}
-
-	var sa = request.storageAllocation()
+	sa := req.storageAllocation()
+	var gbSize = sizeInGB(sa.bSize())
+	var minLockDemand state.Balance
 
-	blobberNodes, bSize, err := ssc.selectBlobbers(
-		creationDate, *allBlobbersList, sa, creationDate.Unix(), balances)
-	if err != nil {
-		return "", common.NewErrInternal("selecting blobbers", err.Error())
+	ids := append(req.Blobbers, blobbers...)
```

Comment:
Append the blobber ids to req.Blobbers? Duplicate ids could happen if the blobbers contains IDs that are in req.Blobbers too.